## Chapter 4 - Bài  Phân tích dữ liệu shop online 

#### Bạn sẽ là nhà phân tích dữ liệu cho một shop giày trực tuyến (ShoeFly.com). Dữ liệu được lưu vào file shoefly.csv

In [265]:
import pandas as pd

In [266]:
# Đọc file dữ liệu đã cho shoefly.csv , và lưu vào biến orders
orders = pd.read_csv('shoefly.csv')

In [267]:
# Hiển thị 5 dòng đầu tiên của orders
orders.head()

order_id first_name last_name  gender                         email  \
0     54791    Rebecca   Lindsay  female  RebeccaLindsay57@hotmail.com   
1     53450      Emily     Joyce  female        EmilyJoyce25@gmail.com   
2     91987      Joyce    Waller  female        Joyce.Waller@gmail.com   
3     14437     Justin  Erickson    male   Justin.Erickson@outlook.com   
4     79357     Andrew     Banks    male              AB4318@gmail.com   

      shoe_type shoe_material shoe_color  price  amount  
0         clogs  faux-leather      black     75       1  
1  ballet flats  faux-leather       navy     36       1  
2       sandles        fabric      black     55       1  
3         clogs  faux-leather        red     49       2  
4         boots       leather      brown     59       1

<details>
  <summary>Nhấn vào đây để xem kết quả!</summary>
  <div>
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>order_id</th>
      <th>first_name</th>
      <th>last_name</th>
      <th>gender</th>
      <th>email</th>
      <th>shoe_type</th>
      <th>shoe_material</th>
      <th>shoe_color</th>
      <th>price</th>
      <th>amount</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>54791</td>
      <td>Rebecca</td>
      <td>Lindsay</td>
      <td>female</td>
      <td>RebeccaLindsay57@hotmail.com</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>black</td>
      <td>75</td>
      <td>1</td>
    </tr>
    <tr>
      <td>1</td>
      <td>53450</td>
      <td>Emily</td>
      <td>Joyce</td>
      <td>female</td>
      <td>EmilyJoyce25@gmail.com</td>
      <td>ballet flats</td>
      <td>faux-leather</td>
      <td>navy</td>
      <td>36</td>
      <td>1</td>
    </tr>
    <tr>
      <td>2</td>
      <td>91987</td>
      <td>Joyce</td>
      <td>Waller</td>
      <td>female</td>
      <td>Joyce.Waller@gmail.com</td>
      <td>sandles</td>
      <td>fabric</td>
      <td>black</td>
      <td>55</td>
      <td>1</td>
    </tr>
    <tr>
      <td>3</td>
      <td>14437</td>
      <td>Justin</td>
      <td>Erickson</td>
      <td>male</td>
      <td>Justin.Erickson@outlook.com</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>red</td>
      <td>49</td>
      <td>2</td>
    </tr>
    <tr>
      <td>4</td>
      <td>79357</td>
      <td>Andrew</td>
      <td>Banks</td>
      <td>male</td>
      <td>AB4318@gmail.com</td>
      <td>boots</td>
      <td>leather</td>
      <td>brown</td>
      <td>59</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
</div></div>
  
</details>

In [268]:
# Hiển thị thông tin của orders
orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   order_id       20 non-null     int64 
 1   first_name     20 non-null     object
 2   last_name      20 non-null     object
 3   gender         20 non-null     object
 4   email          20 non-null     object
 5   shoe_type      20 non-null     object
 6   shoe_material  20 non-null     object
 7   shoe_color     20 non-null     object
 8   price          20 non-null     int64 
 9   amount         20 non-null     int64 
dtypes: int64(3), object(7)
memory usage: 1.7+ KB


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
<pre>&lt;class 'pandas.core.frame.DataFrame'&gt;
RangeIndex: 20 entries, 0 to 19
Data columns (total 13 columns):
order_id         20 non-null int64
first_name       20 non-null object
last_name        20 non-null object
gender           20 non-null object
email            20 non-null object
shoe_type        20 non-null object
shoe_material    20 non-null object
shoe_color       20 non-null object
price            20 non-null int64
amount           20 non-null int64
total            20 non-null int64
shoe_source      20 non-null object
salutation       20 non-null object
dtypes: int64(4), object(9)
memory usage: 2.2+ KB
</pre>
</details>

In [269]:
# 1. Hiển thị và cho biết nhận xét về thông tin thống kê chung của price và amount
orders[['price','amount']].describe()

price     amount
count   20.000000  20.000000
mean    56.650000   1.300000
std     17.011683   0.571241
min     36.000000   1.000000
25%     45.000000   1.000000
50%     55.000000   1.000000
75%     63.500000   1.250000
max    105.000000   3.000000

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
<pre>     
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>price</th>
      <th>amount</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>count</td>
      <td>20.000000</td>
      <td>20.000000</td>
    </tr>
    <tr>
      <td>mean</td>
      <td>56.650000</td>
      <td>1.300000</td>
    </tr>
    <tr>
      <td>std</td>
      <td>17.011683</td>
      <td>0.571241</td>
    </tr>
    <tr>
      <td>min</td>
      <td>36.000000</td>
      <td>1.000000</td>
    </tr>
    <tr>
      <td>25%</td>
      <td>45.000000</td>
      <td>1.000000</td>
    </tr>
    <tr>
      <td>50%</td>
      <td>55.000000</td>
      <td>1.000000</td>
    </tr>
    <tr>
      <td>75%</td>
      <td>63.500000</td>
      <td>1.250000</td>
    </tr>
    <tr>
      <td>max</td>
      <td>105.000000</td>
      <td>3.000000</td>
    </tr>
  </tbody>
</table>
</div></div>
</pre>

</details>

In [270]:
#2. Thêm 1 cột total = price * amount
orders['total'] = orders['price']*orders['amount']
orders

order_id  first_name   last_name  gender                         email  \
0      54791     Rebecca     Lindsay  female  RebeccaLindsay57@hotmail.com   
1      53450       Emily       Joyce  female        EmilyJoyce25@gmail.com   
2      91987       Joyce      Waller  female        Joyce.Waller@gmail.com   
3      14437      Justin    Erickson    male   Justin.Erickson@outlook.com   
4      79357      Andrew       Banks    male              AB4318@gmail.com   
5      52386       Julie       Marsh  female        JulieMarsh59@gmail.com   
6      20487      Thomas      Jensen    male              TJ5470@gmail.com   
7      76971      Janice       Hicks  female        Janice.Hicks@gmail.com   
8      21586     Gabriel      Porter    male     GabrielPorter24@gmail.com   
9      62083     Frances      Palmer  female     FrancesPalmer50@gmail.com   
10     91629     Jessica        Hale  female       JessicaHale25@gmail.com   
11     98602    Lawrence      Parker    male    LawrenceParker44@gmail.com   
12     45832       Susan      Dennis  female       SusanDennis58@gmail.com   
13     33862       Diane       Ochoa  female              DO2680@gmail.com   
14     73431     Rebecca     Charles  female     Rebecca.Charles@gmail.com   
15     93889  Jacqueline       Crane  female            JC2072@hotmail.com   
16     39888     Vincent  Stephenson    male            VS4753@outlook.com   
17     35961         Roy     Tillman    male        RoyTillman20@gmail.com   
18     24560      Thomas    Roberson    male     Thomas.Roberson@gmail.com   
19     28559      Angela      Newton  female       ANewton1977@outlook.com   

       shoe_type shoe_material shoe_color  price  amount  total  
0          clogs  faux-leather      black     75       1     75  
1   ballet flats  faux-leather       navy     36       1     36  
2        sandles        fabric      black     55       1     55  
3          clogs  faux-leather        red     49       2     98  
4          boots       leather      brown     59       1     59  
5        sandles        fabric      black     39       2     78  
6          clogs        fabric       navy     45       1     45  
7          clogs  faux-leather       navy     62       1     62  
8          clogs       leather      brown    105       1    105  
9         wedges       leather      white     36       1     36  
10         clogs       leather        red     72       1     72  
11        wedges        fabric      brown     65       1     65  
12  ballet flats        fabric      white     49       2     98  
13       sandles        fabric        red     55       1     55  
14         boots  faux-leather      white     45       2     90  
15        wedges        fabric        red     79       1     79  
16         boots       leather      black     39       3    117  
17         boots       leather      white     48       1     48  
18        wedges        fabric        red     57       1     57  
19        wedges        fabric        red     63       1     63

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
<pre>     
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>order_id</th>
      <th>shoe_type</th>
      <th>shoe_material</th>
      <th>shoe_color</th>
      <th>price</th>
      <th>amount</th>
      <th>total</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>54791</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>black</td>
      <td>75</td>
      <td>1</td>
      <td>75</td>
    </tr>
    <tr>
      <td>1</td>
      <td>53450</td>
      <td>ballet flats</td>
      <td>faux-leather</td>
      <td>navy</td>
      <td>36</td>
      <td>1</td>
      <td>36</td>
    </tr>
    <tr>
      <td>2</td>
      <td>91987</td>
      <td>sandles</td>
      <td>fabric</td>
      <td>black</td>
      <td>55</td>
      <td>1</td>
      <td>55</td>
    </tr>
    <tr>
      <td>3</td>
      <td>14437</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>red</td>
      <td>49</td>
      <td>2</td>
      <td>98</td>
    </tr>
    <tr>
      <td>4</td>
      <td>79357</td>
      <td>boots</td>
      <td>leather</td>
      <td>brown</td>
      <td>59</td>
      <td>1</td>
      <td>59</td>
    </tr>
  </tbody>
</table>
</div></div></pre>

</details>

In [271]:
#2. Để phục vụ cho chiến dịch sản phẩm giày faux-leather mới, bộ phận marketing muốn gửi email giới thiệu sản phẩm mới 
#  đến tất cả những người đã từng mua giày faux-leather
orders[orders['shoe_material']=='faux-leather'].email

0     RebeccaLindsay57@hotmail.com
1           EmilyJoyce25@gmail.com
3      Justin.Erickson@outlook.com
7           Janice.Hicks@gmail.com
14       Rebecca.Charles@gmail.com
Name: email, dtype: object

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
<pre>0     RebeccaLindsay57@hotmail.com
1           EmilyJoyce25@gmail.com
3      Justin.Erickson@outlook.com
7           Janice.Hicks@gmail.com
14       Rebecca.Charles@gmail.com
Name: email, dtype: object</pre>
</details>

In [272]:
#3. Frances Palmer nói rằng đơn đặt hàng của cô đã sai. Vậy, Frances Palmer đã đặt hàng gì?
orders.loc[(orders['first_name']=='Frances') & (orders['last_name']=='Palmer')]

order_id first_name last_name  gender                      email shoe_type  \
9     62083    Frances    Palmer  female  FrancesPalmer50@gmail.com    wedges   

  shoe_material shoe_color  price  amount  total  
9       leather      white     36       1     36

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>order_id</th>
      <th>first_name</th>
      <th>last_name</th>
      <th>gender</th>
      <th>email</th>
      <th>shoe_type</th>
      <th>shoe_material</th>
      <th>shoe_color</th>
      <th>price</th>
      <th>amount</th>
      <th>total</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>9</td>
      <td>62083</td>
      <td>Frances</td>
      <td>Palmer</td>
      <td>female</td>
      <td>FrancesPalmer50@gmail.com</td>
      <td>wedges</td>
      <td>leather</td>
      <td>white</td>
      <td>36</td>
      <td>1</td>
      <td>36</td>
    </tr>
  </tbody>
</table>
</div></div>

</details>

In [273]:
#4.Hãy liệt kê tất cả các đơn đặt hàng có loại giày là clogs, boots, và ballet flats
orders.loc[(orders['shoe_type']=='clogs')|(orders['shoe_type']=='boots')|(orders['shoe_type']=='ballet flats')]

order_id first_name   last_name  gender                         email  \
0      54791    Rebecca     Lindsay  female  RebeccaLindsay57@hotmail.com   
1      53450      Emily       Joyce  female        EmilyJoyce25@gmail.com   
3      14437     Justin    Erickson    male   Justin.Erickson@outlook.com   
4      79357     Andrew       Banks    male              AB4318@gmail.com   
6      20487     Thomas      Jensen    male              TJ5470@gmail.com   
7      76971     Janice       Hicks  female        Janice.Hicks@gmail.com   
8      21586    Gabriel      Porter    male     GabrielPorter24@gmail.com   
10     91629    Jessica        Hale  female       JessicaHale25@gmail.com   
12     45832      Susan      Dennis  female       SusanDennis58@gmail.com   
14     73431    Rebecca     Charles  female     Rebecca.Charles@gmail.com   
16     39888    Vincent  Stephenson    male            VS4753@outlook.com   
17     35961        Roy     Tillman    male        RoyTillman20@gmail.com   

       shoe_type shoe_material shoe_color  price  amount  total  
0          clogs  faux-leather      black     75       1     75  
1   ballet flats  faux-leather       navy     36       1     36  
3          clogs  faux-leather        red     49       2     98  
4          boots       leather      brown     59       1     59  
6          clogs        fabric       navy     45       1     45  
7          clogs  faux-leather       navy     62       1     62  
8          clogs       leather      brown    105       1    105  
10         clogs       leather        red     72       1     72  
12  ballet flats        fabric      white     49       2     98  
14         boots  faux-leather      white     45       2     90  
16         boots       leather      black     39       3    117  
17         boots       leather      white     48       1     48

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>order_id</th>
      <th>first_name</th>
      <th>last_name</th>
      <th>gender</th>
      <th>email</th>
      <th>shoe_type</th>
      <th>shoe_material</th>
      <th>shoe_color</th>
      <th>price</th>
      <th>amount</th>
      <th>total</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>54791</td>
      <td>Rebecca</td>
      <td>Lindsay</td>
      <td>female</td>
      <td>RebeccaLindsay57@hotmail.com</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>black</td>
      <td>75</td>
      <td>1</td>
      <td>75</td>
    </tr>
    <tr>
      <td>1</td>
      <td>53450</td>
      <td>Emily</td>
      <td>Joyce</td>
      <td>female</td>
      <td>EmilyJoyce25@gmail.com</td>
      <td>ballet flats</td>
      <td>faux-leather</td>
      <td>navy</td>
      <td>36</td>
      <td>1</td>
      <td>36</td>
    </tr>
    <tr>
      <td>3</td>
      <td>14437</td>
      <td>Justin</td>
      <td>Erickson</td>
      <td>male</td>
      <td>Justin.Erickson@outlook.com</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>red</td>
      <td>49</td>
      <td>2</td>
      <td>98</td>
    </tr>
    <tr>
      <td>4</td>
      <td>79357</td>
      <td>Andrew</td>
      <td>Banks</td>
      <td>male</td>
      <td>AB4318@gmail.com</td>
      <td>boots</td>
      <td>leather</td>
      <td>brown</td>
      <td>59</td>
      <td>1</td>
      <td>59</td>
    </tr>
    <tr>
      <td>6</td>
      <td>20487</td>
      <td>Thomas</td>
      <td>Jensen</td>
      <td>male</td>
      <td>TJ5470@gmail.com</td>
      <td>clogs</td>
      <td>fabric</td>
      <td>navy</td>
      <td>45</td>
      <td>1</td>
      <td>45</td>
    </tr>
  </tbody>
</table>
</div></div>
</details>

In [274]:
# 5. Nhiều khách hàng muốn mua loại giày mà làm từ vật liệu không phải nguồn gốc là da động vật. 
# Hãy thêm một cột mới shoe_source, nếu các vật liệu là leather thì là animal, ngược lại thì ghi là vegan 
orders.loc[orders['shoe_material'] != 'leather','shoe_source']='vegan'
orders.loc[orders['shoe_material'] == 'leather','shoe_source']='animal'
orders

order_id  first_name   last_name  gender                         email  \
0      54791     Rebecca     Lindsay  female  RebeccaLindsay57@hotmail.com   
1      53450       Emily       Joyce  female        EmilyJoyce25@gmail.com   
2      91987       Joyce      Waller  female        Joyce.Waller@gmail.com   
3      14437      Justin    Erickson    male   Justin.Erickson@outlook.com   
4      79357      Andrew       Banks    male              AB4318@gmail.com   
5      52386       Julie       Marsh  female        JulieMarsh59@gmail.com   
6      20487      Thomas      Jensen    male              TJ5470@gmail.com   
7      76971      Janice       Hicks  female        Janice.Hicks@gmail.com   
8      21586     Gabriel      Porter    male     GabrielPorter24@gmail.com   
9      62083     Frances      Palmer  female     FrancesPalmer50@gmail.com   
10     91629     Jessica        Hale  female       JessicaHale25@gmail.com   
11     98602    Lawrence      Parker    male    LawrenceParker44@gmail.com   
12     45832       Susan      Dennis  female       SusanDennis58@gmail.com   
13     33862       Diane       Ochoa  female              DO2680@gmail.com   
14     73431     Rebecca     Charles  female     Rebecca.Charles@gmail.com   
15     93889  Jacqueline       Crane  female            JC2072@hotmail.com   
16     39888     Vincent  Stephenson    male            VS4753@outlook.com   
17     35961         Roy     Tillman    male        RoyTillman20@gmail.com   
18     24560      Thomas    Roberson    male     Thomas.Roberson@gmail.com   
19     28559      Angela      Newton  female       ANewton1977@outlook.com   

       shoe_type shoe_material shoe_color  price  amount  total shoe_source  
0          clogs  faux-leather      black     75       1     75       vegan  
1   ballet flats  faux-leather       navy     36       1     36       vegan  
2        sandles        fabric      black     55       1     55       vegan  
3          clogs  faux-leather        red     49       2     98       vegan  
4          boots       leather      brown     59       1     59      animal  
5        sandles        fabric      black     39       2     78       vegan  
6          clogs        fabric       navy     45       1     45       vegan  
7          clogs  faux-leather       navy     62       1     62       vegan  
8          clogs       leather      brown    105       1    105      animal  
9         wedges       leather      white     36       1     36      animal  
10         clogs       leather        red     72       1     72      animal  
11        wedges        fabric      brown     65       1     65       vegan  
12  ballet flats        fabric      white     49       2     98       vegan  
13       sandles        fabric        red     55       1     55       vegan  
14         boots  faux-leather      white     45       2     90       vegan  
15        wedges        fabric        red     79       1     79       vegan  
16         boots       leather      black     39       3    117      animal  
17         boots       leather      white     48       1     48      animal  
18        wedges        fabric        red     57       1     57       vegan  
19        wedges        fabric        red     63       1     63       vegan

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>first_name</th>
      <th>last_name</th>
      <th>shoe_type</th>
      <th>shoe_material</th>
      <th>shoe_color</th>
      <th>shoe_source</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>Rebecca</td>
      <td>Lindsay</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>black</td>
      <td>vegan</td>
    </tr>
    <tr>
      <td>1</td>
      <td>Emily</td>
      <td>Joyce</td>
      <td>ballet flats</td>
      <td>faux-leather</td>
      <td>navy</td>
      <td>vegan</td>
    </tr>
    <tr>
      <td>2</td>
      <td>Joyce</td>
      <td>Waller</td>
      <td>sandles</td>
      <td>fabric</td>
      <td>black</td>
      <td>vegan</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Justin</td>
      <td>Erickson</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>red</td>
      <td>vegan</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Andrew</td>
      <td>Banks</td>
      <td>boots</td>
      <td>leather</td>
      <td>brown</td>
      <td>animal</td>
    </tr>
  </tbody>
</table>
</div></div>
</details>

In [275]:
# 6. Bộ phận marketting muốn gửi email đến từng khách hàng.
# Sử dụng các cột last_name và gender để tạo ra một cột mới là salutation:
# là Dear Mr. <last_name> nếu KH là nam và Dear Ms. <last_name> nếu KH là nữ.

orders.loc[orders['gender']=='female','salutation']= 'Dear Ms. ' + orders['last_name']
orders.loc[orders['gender']!='female','salutation']= 'Dear Mr. ' + orders['last_name']

orders

order_id  first_name   last_name  gender                         email  \
0      54791     Rebecca     Lindsay  female  RebeccaLindsay57@hotmail.com   
1      53450       Emily       Joyce  female        EmilyJoyce25@gmail.com   
2      91987       Joyce      Waller  female        Joyce.Waller@gmail.com   
3      14437      Justin    Erickson    male   Justin.Erickson@outlook.com   
4      79357      Andrew       Banks    male              AB4318@gmail.com   
5      52386       Julie       Marsh  female        JulieMarsh59@gmail.com   
6      20487      Thomas      Jensen    male              TJ5470@gmail.com   
7      76971      Janice       Hicks  female        Janice.Hicks@gmail.com   
8      21586     Gabriel      Porter    male     GabrielPorter24@gmail.com   
9      62083     Frances      Palmer  female     FrancesPalmer50@gmail.com   
10     91629     Jessica        Hale  female       JessicaHale25@gmail.com   
11     98602    Lawrence      Parker    male    LawrenceParker44@gmail.com   
12     45832       Susan      Dennis  female       SusanDennis58@gmail.com   
13     33862       Diane       Ochoa  female              DO2680@gmail.com   
14     73431     Rebecca     Charles  female     Rebecca.Charles@gmail.com   
15     93889  Jacqueline       Crane  female            JC2072@hotmail.com   
16     39888     Vincent  Stephenson    male            VS4753@outlook.com   
17     35961         Roy     Tillman    male        RoyTillman20@gmail.com   
18     24560      Thomas    Roberson    male     Thomas.Roberson@gmail.com   
19     28559      Angela      Newton  female       ANewton1977@outlook.com   

       shoe_type shoe_material shoe_color  price  amount  total shoe_source  \
0          clogs  faux-leather      black     75       1     75       vegan   
1   ballet flats  faux-leather       navy     36       1     36       vegan   
2        sandles        fabric      black     55       1     55       vegan   
3          clogs  faux-leather        red     49       2     98       vegan   
4          boots       leather      brown     59       1     59      animal   
5        sandles        fabric      black     39       2     78       vegan   
6          clogs        fabric       navy     45       1     45       vegan   
7          clogs  faux-leather       navy     62       1     62       vegan   
8          clogs       leather      brown    105       1    105      animal   
9         wedges       leather      white     36       1     36      animal   
10         clogs       leather        red     72       1     72      animal   
11        wedges        fabric      brown     65       1     65       vegan   
12  ballet flats        fabric      white     49       2     98       vegan   
13       sandles        fabric        red     55       1     55       vegan   
14         boots  faux-leather      white     45       2     90       vegan   
15        wedges        fabric        red     79       1     79       vegan   
16         boots       leather      black     39       3    117      animal   
17         boots       leather      white     48       1     48      animal   
18        wedges        fabric        red     57       1     57       vegan   
19        wedges        fabric        red     63       1     63       vegan   

             salutation  
0      Dear Ms. Lindsay  
1        Dear Ms. Joyce  
2       Dear Ms. Waller  
3     Dear Mr. Erickson  
4        Dear Mr. Banks  
5        Dear Ms. Marsh  
6       Dear Mr. Jensen  
7        Dear Ms. Hicks  
8       Dear Mr. Porter  
9       Dear Ms. Palmer  
10        Dear Ms. Hale  
11      Dear Mr. Parker  
12      Dear Ms. Dennis  
13       Dear Ms. Ochoa  
14     Dear Ms. Charles  
15       Dear Ms. Crane  
16  Dear Mr. Stephenson  
17     Dear Mr. Tillman  
18    Dear Mr. Roberson  
19      Dear Ms. Newton

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>first_name</th>
      <th>last_name</th>
      <th>gender</th>
      <th>salutation</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>Rebecca</td>
      <td>Lindsay</td>
      <td>female</td>
      <td>Dear Ms. Lindsay</td>
    </tr>
    <tr>
      <td>1</td>
      <td>Emily</td>
      <td>Joyce</td>
      <td>female</td>
      <td>Dear Ms. Joyce</td>
    </tr>
    <tr>
      <td>2</td>
      <td>Joyce</td>
      <td>Waller</td>
      <td>female</td>
      <td>Dear Ms. Waller</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Justin</td>
      <td>Erickson</td>
      <td>male</td>
      <td>Dear Mr. Erickson</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Andrew</td>
      <td>Banks</td>
      <td>male</td>
      <td>Dear Mr. Banks</td>
    </tr>
  </tbody>
</table>
</div></div>
</details>

In [276]:
#7. Hãy cho biết giá trị đơn hàng trung bình của các mặt hàng giày thuộc nhóm animal là bao nhiêu
orders.groupby('shoe_source').get_group('animal').total.mean().round(2)

72.83

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
72.83
</details>

In [277]:
#8. Hiển thị orders theo thứ tự giảm dần total và tăng dần last_name

orders.sort_values(['total','last_name'],ascending=[False,True])

order_id  first_name   last_name  gender                         email  \
16     39888     Vincent  Stephenson    male            VS4753@outlook.com   
8      21586     Gabriel      Porter    male     GabrielPorter24@gmail.com   
12     45832       Susan      Dennis  female       SusanDennis58@gmail.com   
3      14437      Justin    Erickson    male   Justin.Erickson@outlook.com   
14     73431     Rebecca     Charles  female     Rebecca.Charles@gmail.com   
15     93889  Jacqueline       Crane  female            JC2072@hotmail.com   
5      52386       Julie       Marsh  female        JulieMarsh59@gmail.com   
0      54791     Rebecca     Lindsay  female  RebeccaLindsay57@hotmail.com   
10     91629     Jessica        Hale  female       JessicaHale25@gmail.com   
11     98602    Lawrence      Parker    male    LawrenceParker44@gmail.com   
19     28559      Angela      Newton  female       ANewton1977@outlook.com   
7      76971      Janice       Hicks  female        Janice.Hicks@gmail.com   
4      79357      Andrew       Banks    male              AB4318@gmail.com   
18     24560      Thomas    Roberson    male     Thomas.Roberson@gmail.com   
13     33862       Diane       Ochoa  female              DO2680@gmail.com   
2      91987       Joyce      Waller  female        Joyce.Waller@gmail.com   
17     35961         Roy     Tillman    male        RoyTillman20@gmail.com   
6      20487      Thomas      Jensen    male              TJ5470@gmail.com   
1      53450       Emily       Joyce  female        EmilyJoyce25@gmail.com   
9      62083     Frances      Palmer  female     FrancesPalmer50@gmail.com   

       shoe_type shoe_material shoe_color  price  amount  total shoe_source  \
16         boots       leather      black     39       3    117      animal   
8          clogs       leather      brown    105       1    105      animal   
12  ballet flats        fabric      white     49       2     98       vegan   
3          clogs  faux-leather        red     49       2     98       vegan   
14         boots  faux-leather      white     45       2     90       vegan   
15        wedges        fabric        red     79       1     79       vegan   
5        sandles        fabric      black     39       2     78       vegan   
0          clogs  faux-leather      black     75       1     75       vegan   
10         clogs       leather        red     72       1     72      animal   
11        wedges        fabric      brown     65       1     65       vegan   
19        wedges        fabric        red     63       1     63       vegan   
7          clogs  faux-leather       navy     62       1     62       vegan   
4          boots       leather      brown     59       1     59      animal   
18        wedges        fabric        red     57       1     57       vegan   
13       sandles        fabric        red     55       1     55       vegan   
2        sandles        fabric      black     55       1     55       vegan   
17         boots       leather      white     48       1     48      animal   
6          clogs        fabric       navy     45       1     45       vegan   
1   ballet flats  faux-leather       navy     36       1     36       vegan   
9         wedges       leather      white     36       1     36      animal   

             salutation  
16  Dear Mr. Stephenson  
8       Dear Mr. Porter  
12      Dear Ms. Dennis  
3     Dear Mr. Erickson  
14     Dear Ms. Charles  
15       Dear Ms. Crane  
5        Dear Ms. Marsh  
0      Dear Ms. Lindsay  
10        Dear Ms. Hale  
11      Dear Mr. Parker  
19      Dear Ms. Newton  
7        Dear Ms. Hicks  
4        Dear Mr. Banks  
18    Dear Mr. Roberson  
13       Dear Ms. Ochoa  
2       Dear Ms. Waller  
17     Dear Mr. Tillman  
6       Dear Mr. Jensen  
1        Dear Ms. Joyce  
9       Dear Ms. Palmer

<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
<div class="output_subarea output_html rendered_html output_result"><div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>first_name</th>
      <th>last_name</th>
      <th>gender</th>
      <th>email</th>
      <th>shoe_type</th>
      <th>shoe_material</th>
      <th>shoe_color</th>
      <th>price</th>
      <th>amount</th>
      <th>total</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>16</td>
      <td>Vincent</td>
      <td>Stephenson</td>
      <td>male</td>
      <td>VS4753@outlook.com</td>
      <td>boots</td>
      <td>leather</td>
      <td>black</td>
      <td>39</td>
      <td>3</td>
      <td>117</td>
    </tr>
    <tr>
      <td>8</td>
      <td>Gabriel</td>
      <td>Porter</td>
      <td>male</td>
      <td>GabrielPorter24@gmail.com</td>
      <td>clogs</td>
      <td>leather</td>
      <td>brown</td>
      <td>105</td>
      <td>1</td>
      <td>105</td>
    </tr>
    <tr>
      <td>12</td>
      <td>Susan</td>
      <td>Dennis</td>
      <td>female</td>
      <td>SusanDennis58@gmail.com</td>
      <td>ballet flats</td>
      <td>fabric</td>
      <td>white</td>
      <td>49</td>
      <td>2</td>
      <td>98</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Justin</td>
      <td>Erickson</td>
      <td>male</td>
      <td>Justin.Erickson@outlook.com</td>
      <td>clogs</td>
      <td>faux-leather</td>
      <td>red</td>
      <td>49</td>
      <td>2</td>
      <td>98</td>
    </tr>
    <tr>
      <td>14</td>
      <td>Rebecca</td>
      <td>Charles</td>
      <td>female</td>
      <td>Rebecca.Charles@gmail.com</td>
      <td>boots</td>
      <td>faux-leather</td>
      <td>white</td>
      <td>45</td>
      <td>2</td>
      <td>90</td>
    </tr>
  </tbody>
</table>
</div></div>
</details>